monitor/metrics/util.go

```diff
@@ -14,13 +14,22 @@ func PeersetFilter(metrics []*api.Metric, peerset []peer.ID) []*api.Metric {
 		peerMap[pid] = struct{}{}
 	}
 
+	metricsByPeer := make(map[peer.ID]*api.Metric)
 	filtered := make([]*api.Metric, 0, len(metrics))
 
 	for _, metric := range metrics {
 		_, ok := peerMap[metric.Peer]
 		if !ok {
 			continue
 		}
+		metricsByPeer[metric.Peer] = metric
+	}
+
+	for _, peer := range peerset {
+		metric, ok := metricsByPeer[peer]
+		if !ok {
+			continue
+		}
```

Comment:
This is changing behavior. The function now only allows 1 metric per peer. It is used only in that case, but it should be reflected in the godocs that this function is not doing that it says it's doing anymore.

Also, if doing it this way, then re-use peerMap to store the metrics rather than adding another map with peers as keys.